src/storage/lock_manager/waiter_manager.rs

```diff
@@ -177,17 +176,22 @@ impl Scheduler {
         cb: StorageCb,
         pr: ProcessResult,
         lock: Lock,
+        ttl: u64,
         is_first_lock: bool,
-        timeout: u64,
     ) {
-        self.notify_scheduler(Task::WaitFor {
-            start_ts,
-            cb,
-            pr,
-            lock,
-            is_first_lock,
-            timeout,
-        });
+        let timeout = calculate_wait_for_lock_timeout(lock.ts, ttl);
```

Comment:
Why use `ts` and `ttl` to calculate timeout?

Reply:
Now the timeout is `min(max_wait_for_lock_timeout, now - (lock's start_ts + lock's ttl))`. If the lock has expired, txns won't wait for it.

Reply:
We can afford to wait for an expired lock for simpler code.